tpvasconcelos/ridgeplot
# feat: add automatic RST generator for internal API docs

This PR adds functionality to automatically generate .rst files for internal API documentation.

Changes:
- Add `generate_internal_api_rst.py` script that:
  - Automatically detects internal modules
  - Generates appropriate RST files with consistent formatting
  - Maintains hierarchical module structure
  - Provides meaningful module descriptions
- Add tests for core functionality
- Keep consistent docstring formats

The script generates RST files in `docs/api/internal/` with proper module
hierarchy and descriptions, removing the need for manual maintenance.

Testing:
1. Run the script:
```bash
python cicd_utils/cicd/scripts/generate_internal_api_rst.py
```
2. Check generated files in  `docs/api/internal/`

3. Run tests:
```bash
tox -e pytest -- tests/cicd_utils/test_scripts/test_generate_internal_api_rst.py
```

Note: The script can be integrated into the CI pipeline in a future iteration if needed.

## PR check list



- [ ] Read the [contributing guidelines](https://ridgeplot.readthedocs.io/en/latest/development/contributing.html).
  - You don't have to read the whole thing, but it's a good idea to skim it. And definitely take a look at it if you're experiencing any issues getting your local environment up and running.
- [ ] Provided the relevant details in the PR's description.
- [ ] Referenced relevant issues in the PR's description.
- [ ] Added tests for all my changes.
  - The CI will fail unless 100% of all new code is covered by the tests.
- [ ] Updated the docs for relevant changes.
  - [ ] New modules (or renamed ones) are included in `docs/api/internal/`.
  - [ ] New public functions/classes/variables are documented in `docs/api/index.rst`.
  - [ ] Added the appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded) to docstrings.
    - The version should be the next release version, which you can infer by bumping the minor version in `MAJOR.MINOR.PATCH` (e.g., if the current version is `0.2.3`, the next release will be `0.3.0`).
- [ ] Changes are documented in `docs/reference/changelog.md`.
  - Please try to follow the conventions laid out in the file. In doubt, just ask!
- [ ] Consider granting [push permissions to your PR's branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so maintainers can help you out if needed.
- [ ] The CI check are all passing, or I'm working on fixing them!



----
📚 Documentation preview 📚: https://ridgeplot--293.org.readthedocs.build/en/293/



Closes #290
